File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.js
/**
 * @param {string} s
 * @return {number}
 */
var maxFreqSum = function(s) {

    let map = {};

    for(let i=0;i<s.length;i++){
        if(!map[s[i]]){
            map[s[i]]=1;
        }else{
            ++map[s[i]];
        }

    }

    // find the max vowel and consonant inside the map.
    let vowels = ['a','e','i','o','u'];
    let maxVowel = 0;
    let maxConsonant = 0;

    for(let i=0;i<s.length;i++){
        if(vowels.includes(s[i])){
           maxVowel = Math.max(maxVowel, map[s[i]]);
           
        }else{
            maxConsonant = Math.max(maxConsonant,map[s[i]]);
           
        }
    }
    return maxVowel + maxConsonant;
    
};

File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.java
class Solution {
    public int maxFreqSum(String s) {

        HashMap<Character,Integer> map = new HashMap<>();

        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            if(!map.containsKey(c)){
                map.put(c, 1);
            }else{
               map.put(c,map.get(c)+1);
            }
        }

         Set<Character> vowels = new HashSet<>(Arrays.asList('a','e','i','o','u'));


        int maxVowel = 0;
        int maxConsonant = 0;

        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            if(vowels.contains(c)){
                maxVowel = Math.max(maxVowel, map.get(c));
            }else{
                maxConsonant = Math.max(maxConsonant ,  map.get(c));
            }
        }
        return maxVowel + maxConsonant;
        
    }
}